nav2_amcl/CMakeLists.txt

```diff
@@ -89,7 +90,7 @@ ament_target_dependencies(${library_name}
 )
 
 target_link_libraries(${library_name}
-  map_lib motions_lib sensors_lib motion_models
+  map_lib sensors_lib motion_models
```

Comment:
I'm almost entirely sure that `motion_models` shouldn't be linked here, that was the crux of the issue, the pluginlib library should not be linked against the use library.